subprojects/core/src/integTest/groovy/org/gradle/api/BuildScriptErrorIntegrationTest.groovy

```diff
@@ -166,6 +166,6 @@ def doSomething() {
 
         expect:
         fails()
-        failure.assertHasCause("Could not find method iDontExist() for arguments [foo-bar] on object of type org.gradle.api.plugins.jvm.internal.DefaultJvmComponentDependencies.")
+        failure.assertHasCause("Could not find method iDontExist() for arguments [foo-bar] on property 'dependencies' of type java.lang.Object.")
```

Comment:
🤔 Hrm, where is this Object coming from?

Reply:
Comes from `getSuperclass` as you changed here https://github.com/gradle/gradle/pull/25242/files#diff-a135921792daf52c5d8258172d93c9c675260977ce57409ff39fba4513f1a975

Reply:
I'd like to change this in another PR if possible (Can do it right after this one) in case it affects error messages everywhere

Reply:
SGTM

Reply:
Filed this issue: https://github.com/gradle/gradle/issues/27438